src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -54,134 +51,42 @@ typedef struct asym_check_region {
 
 static asym_check_region *asym_check_region_p = NULL;
 
-/* MPIDU_shm_seg_alloc(len, ptr_p)
-
-   This function is used to allow the caller to reserve a len sized
-   region in the shared memory segment.  Once the shared memory
-   segment is actually allocated, when MPIDU_SHM_Seg_commit() is
-   called, the pointer *ptr_p will be set to point to the reserved
-   region in the shared memory segment.
-
-   Note that no shared memory is actually allocated by this function,
-   and the *ptr_p pointer will be valid only after
-   MPIDU_SHM_Seg_commit() is called.
-*/
-int MPIDU_shm_seg_alloc(size_t len, void **ptr_p, MPL_memory_class class)
-{
-    int mpi_errno = MPI_SUCCESS;
-    alloc_elem_t *ep;
-    MPIR_CHKPMEM_DECL(1);
-    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDU_SHM_SEG_ALLOC);
-
-    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDU_SHM_SEG_ALLOC);
-
-    /* round up to multiple of 8 to ensure the start of the next
-     * region is 64-bit aligned. */
-    len = MPL_ROUND_UP_ALIGN(len, (size_t) 8);
-    MPIR_Assert(len);
-    MPIR_Assert(ptr_p);
-
-    MPIR_CHKPMEM_MALLOC(ep, alloc_elem_t *, sizeof(alloc_elem_t), mpi_errno, "el", class);
-
-    ep->ptr_p = ptr_p;
-    ep->len = len;
-
-    ALLOCQ_ENQUEUE(ep);
-
-    segment_len += len;
-
-  fn_exit:
-    MPIR_CHKPMEM_COMMIT();
-    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDU_SHM_SEG_ALLOC);
-    return mpi_errno;
-  fn_fail:
-    MPIR_CHKPMEM_REAP();
-    goto fn_exit;
-}
-
-/* MPIDU_shm_seg_commit(memory, num_local, local_rank)
+/* MPIDU_shm_seg_alloc(len, baseaddr)
 
-   This function allocates a shared memory segment large enough to
-   hold all of the regions previously requested by calls to
-   MPIDU_shm_seg_alloc().  For each request, this function sets the
-   associated pointer to point to the reserved region in the allocated
-   shared memory segment.
-
-   If there is only one process local to this node, then a shared
-   memory region is not allocated.  Instead, memory is allocated from
-   the heap.
-
-   At least one call to MPIDU_SHM_Seg_alloc() must be made before
-   calling MPIDU_SHM_Seg_commit().
+   This function allocates a shared memory segment
  */
-int MPIDU_shm_seg_commit(MPIDU_shm_seg_t * memory, MPIDU_shm_barrier_t ** barrier,
-                         int num_local, int local_rank, int local_procs_0, int rank,
-                         MPL_memory_class class)
+int MPIDU_shm_seg_alloc(size_t len, void **ptr, MPL_memory_class class)
```

Comment:
I am leaning toward simplify the interface by assuming the memory class to be `MPL_MEM_SHMEM` (or something like that).

Reply:
Yes I remember this discussion, if nobody has complains I will get rid of `class`.